src/Evolution/Systems/Cce/WorldtubeInterfaceManager.hpp

```diff
@@ -0,0 +1,144 @@
+      noexcept = 0;
+
+  virtual void insert_gh_data(
+      const TimeStepId& time_id,
+      const tnsr::aa<DataVector, 3>& spacetime_metric,
+      const tnsr::iaa<DataVector, 3>& phi, const tnsr::aa<DataVector, 3>& pi,
+      const tnsr::aa<DataVector, 3>& dt_spacetime_metric,
+      const tnsr::iaa<DataVector, 3>& dt_phi,
+      const tnsr::aa<DataVector, 3>& dt_pi) noexcept = 0;
+
+  virtual void request_gh_data(const TimeStepId&) noexcept = 0;
+
+  virtual auto try_retrieve_first_ready_gh_data() noexcept -> boost::optional<
+      std::tuple<TimeStepId, tnsr::aa<DataVector, 3>, tnsr::iaa<DataVector, 3>,
+                 tnsr::aa<DataVector, 3>>> = 0;
+};
+
+/*!
+ * \brief Simple implementation of a `GHWorldtubeInterfaceManager` that only
+ * provides boundary data on matching `TimeStepId`s
+ *
+ * \details This version of the interface manager assumes that the CCE system
+ * and the generalized harmonic system that it communicates with evolve with an
+ * identical time stepper and on identical time step intervals (they evolve in
+ * 'lock step'). As a result, and to streamline communications, new data is
+ * always immediately 'ready' and requests are ignored to produce the behavior
+ * of just immediately gauge-transforming and sending the data to the CCE
+ * component as soon as it becomes available from the GH system.
+ *
+ * \warning Using this interface manager when the GH components and the CCE
+ * evolution are not identically stepped is considered undefined behavior. The
+ * outcome will likely be that CCE will fail to evolve and the boundary data
+ * will be continually inserted into a rapidly expanding inbox.
+ */
+class GHLockstepInterfaceManager : public GHWorldtubeInterfaceManager {
+ public:
+
+  static constexpr OptionString help{
+    "Pass data between GH and CCE systems on matching timesteps only."};
+
+  using options = tmpl::list<>;
+
+  GHLockstepInterfaceManager() = default;
+
+  explicit GHLockstepInterfaceManager(CkMigrateMessage* /*unused*/) noexcept {}
+
+  WRAPPED_PUPable_decl_template(GHLockstepInterfaceManager);  // NOLINT
+
+  std::unique_ptr<GHWorldtubeInterfaceManager> get_clone() const
+      noexcept override;
+
+  /// \brief Store a provided data set in a `std::deque`. Note that the
+  /// lock-step constraint ensures that only the generalized harmonic variables
+  /// `spacetime_metric`, `phi`, and `pi` are used. The remaining variables are
+  /// accepted to comply with the more general abstract interface.
+  void insert_gh_data(
+      const TimeStepId& time_id,
+      const tnsr::aa<DataVector, 3>& spacetime_metric,
+      const tnsr::iaa<DataVector, 3>& phi, const tnsr::aa<DataVector, 3>& pi,
+      const tnsr::aa<DataVector, 3>& dt_spacetime_metric =
+          tnsr::aa<DataVector, 3>{},
+      const tnsr::iaa<DataVector, 3>& dt_phi = tnsr::iaa<DataVector, 3>{},
+      const tnsr::aa<DataVector, 3>& dt_pi =
+          tnsr::aa<DataVector, 3>{}) noexcept override;
```

Comment:
The implementation for these functions are probably in another commit that I haven't looked at yet, but here's the note: they should be in this commit.